notion/block.py

```diff
@@ -647,6 +655,31 @@ def _str_fields(self):
         return super()._str_fields() + ["title", "collection"]
 
 
+class CollectionViewBlockViews(Children):
+
+    child_list_key = "view_ids"
+
+    def _get_block(self, view_id):
+        return self._parent._client.get_collection_view(view_id, collection=self._parent.collection)
+
+    def add_new(self, view_type="table"):
+        if not self._parent.collection:
+            logging.warning("{} does not have 'collection_id' set; skipping.".format(self))
+            return None
+
+        record_id = self._parent._client.create_record(table="collection_view", parent=self._parent, type=view_type)
+        view = self._parent._client.get_collection_view(record_id, collection=self._parent._collection)
+        view.set("collection_id", self._parent._collection.id)
+        view_ids = self._parent.get(CollectionViewBlockViews.child_list_key, [])
+        view_ids.append(view.id)
+        self._parent.set(CollectionViewBlockViews.child_list_key, view_ids)
+
+        # At this point, the view does not see to be completely initialized yet.
+        # Hack: wait a bit before e.g. setting a query.
+        time.sleep(3)
+        return view
```

Comment:
Still puzzling over this. Could you perhaps open a tech debt issue for looking into it further?

Reply:
https://github.com/jamalex/notion-py/issues/46